src/utils/src/SyslogLogger.c

```diff
@@ -0,0 +1,102 @@
+/**
+ * Kinesis Video Producer Syslog based logger
+ */
+#define LOG_CLASS "SyslogLogger"
+#include <syslog.h>
+#include "Include_i.h"
+
+PSyslogLogger gSyslogLogger = NULL;
+
+VOID syslogLoggerLogPrintFn(UINT32 level, PCHAR tag, PCHAR fmt, ...)
+{
+    va_list valist;
+
+    UNUSED_PARAM(tag);
+
+    if (level >= GET_LOGGER_LOG_LEVEL() && gSyslogLogger != NULL) {
+        // Translate log level from libkvs to syslog
+        switch (level) {
+            case LOG_LEVEL_VERBOSE:
+                level = LOG_DEBUG;
+                break;
+            case LOG_LEVEL_DEBUG:
+                level = LOG_DEBUG;
+                break;
+            case LOG_LEVEL_INFO:
+                level = LOG_INFO;
+                break;
+            case LOG_LEVEL_WARN:
+                level = LOG_WARNING;
+                break;
+            case LOG_LEVEL_ERROR:
+                level = LOG_ERR;
+                break;
+            case LOG_LEVEL_FATAL:
+                level = LOG_EMERG;
+                break;
+            default:
+                // ignore unrecognized log levels
+                return;
+        }
+        
+        // Write to syslog
+        va_start(valist, fmt);
+        vsyslog(level, fmt, valist);
+        va_end(valist);
+    }
+}
+
+STATUS createSyslogLogger(UINT32 facility, BOOL printLog, BOOL setGlobalLogFn, logPrintFunc* pSyslogPrintFn)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    CHK(gSyslogLogger == NULL, retStatus); // dont allocate again if already allocated
+
+    // allocate the struct
+    gSyslogLogger = (PSyslogLogger) MEMALLOC(SIZEOF(SyslogLogger));
+    MEMSET(gSyslogLogger, 0x00, SIZEOF(SyslogLogger));
```

Comment:
Try to use 

CHK(NULL != (gSyslogLogger = (PSyslogLogger) MEMCALLOC(1, SIZEOF(SyslogLogger)));

Reply:
Will do. FYI `FileLogger.c` doesn't have this check -- should it?

Reply:
It certainly should :) Would you mind also adding it there?